crucible-mir/src/Mir/Mir.hs

```diff
@@ -265,7 +279,10 @@ data Collection = Collection {
     _statics   :: !(Map DefId Static),
     _vtables   :: !(Map VtableName Vtable),
     _intrinsics :: !(Map IntrinsicName Intrinsic),
-    _namedTys  :: !(Map TyName Ty),
+    _namedTys  :: !(Map TyName (Ty, Maybe Layout)),
+    -- | Layouts for known types. If the value is 'Nothing' then the type is
+    -- unsized.
+    _layouts   :: !(Map Ty (Maybe Layout)),
```

Comment:
The Haddocks should state that there is a precondition that the entries in this `Map` should precisely correspond to the pairs in the range of `_namedTys` (and that this `Map` only exists for caching purposes).